### PR TITLE
Assignment classes

### PR DESCRIPTION
**Pull request by pedrocamargo**

### CODE REVIEW

**Comment by pedrocamargo** [comment]
This is a really small improvement, but it seemed a little too obvious not to make...

File: aequilibrae/paths/traffic_assignment.py [hunk 1]
```diff
@@ -285,7 +303,7 @@ def set_cores(self, cores: int) -> None:
             cores (:obj:`int`): Number of CPU cores to use
         """
         self.cores = cores
-        if self.classes is not None:
+        if self.classes:
```
Comment (jamiecook) [inline aequilibrae/paths/traffic_assignment.py:306]: if classes is an array a for loop over an empty array does nothing so you don't really need to wrap it in an if statement

**Review by jamiecook — APPROVED** [review]

File: aequilibrae/paths/traffic_assignment.py (hunk 1 above)
Comment (pedrocamargo) [reply aequilibrae/paths/traffic_assignment.py:305]: Setting the number of cores should be inside the if, actually...

Comment (jamiecook) [reply aequilibrae/paths/traffic_assignment.py:305]: ah... another nice trick is to not use `else raise`

Comment (jamiecook) [reply aequilibrae/paths/traffic_assignment.py:305]:
```
        if not self.classes:
            raise Exception('You need load traffic classes before overwriting the number of cores')
      
        # continue as per normal
        self.cores = cores
        for c in self.classes:
            c.results.set_cores(cores)
                c._aon_results.set_cores(cores)
```

Comment (pedrocamargo) [reply aequilibrae/paths/traffic_assignment.py]: Hahahahaha  I thought of the same thing while the tests ran locally... hahahaha

**Pull request merged**